php/WC_Gateway_Blockonomics.php

```diff
@@ -184,52 +162,92 @@ public function check_blockonomics_callback()
         }else if ($uuid){
             $this->redirect_to_template('track.php');
         }
-        $address = isset($_REQUEST["finish_order"]) ? $_REQUEST["finish_order"] : "";
-        if ($address) {
-            $order = $orders[$address];
-            $wc_order = new WC_Order($order['order_id']);
+        $order_id = isset($_REQUEST["finish_order"]) ? $_REQUEST["finish_order"] : "";
+        if ($order_id) {
+            $order = $orders[$order_id];
+            $wc_order = new WC_Order($order_id);
             echo $order['order_id'];
             wp_redirect($wc_order->get_checkout_order_received_url());
             exit();
         }
-        $address = isset($_REQUEST['get_order']) ? $_REQUEST['get_order'] : "";
-        if ($address) {
+        $order_id = isset($_REQUEST['get_order']) ? $_REQUEST['get_order'] : "";
+        $crypto= isset($_REQUEST['crypto']) ? $_REQUEST['crypto'] : "";
+        if ($order_id && $crypto) {
+            include_once 'Blockonomics.php';
+            $blockonomics = new Blockonomics;
+            global $woocommerce;
+            $order = $orders[$order_id];
+            $wc_order = new WC_Order($order_id);
+            if(get_woocommerce_currency() != 'BTC'){
+                $price = $blockonomics->get_price(get_woocommerce_currency(), $crypto);
+                $price = $price * 100/(100+get_option('blockonomics_margin', 0));
+            }else{
+                $price = 1;
+            }
+            $currentAddress = get_post_meta($order_id,"blockonomics_address");
+            if($currentAddress[0]['address'] && $currentAddress[0]['crypto'] == $crypto) {
```

Comment:
`btc_address` and `bch_address` are added to the post for admin. `blockonomics_address` can be removed and check for existing address can use these `btc_address` and `bch_address` post meta.